concrete/blocks/desktop_waiting_for_me/controller.php

```diff
@@ -26,13 +29,48 @@ public function getBlockTypeName()
     public function view()
     {
         $this->requireAsset('core/notification');
+
+        $filterValues = ['' => t('** Show All')];
+
+        $factory = $this->app->make(FilterListFactory::class);
+        $filterList = $factory->createList();
+        $filters = $filterList->getFilters();
+        foreach($filters as $filter) {
+            $filterValues[$filter->getKey()] = $filter->getName();
+        }
+
+        $filter = null;
+        if ($this->request->query->has('filter')) {
+            $filter = $this->request->query->get('filter');
+            $this->set('activeFilter', h($filter));
+        }
+
         $u = new \User();
-        $entityManager = $this->app->make('Doctrine\ORM\EntityManager');
-        $r = $entityManager->getRepository('Concrete\Core\Entity\Notification\NotificationAlert');
-        $alerts = $r->findMyAlerts($u);
+        $list = $this->app->make(AlertList::class, ['user' => $u]);
+        if ($filter) {
+            $activeFilter = $filterList->getFilterByKey($filter);
+            if ($activeFilter) {
+                $activeFilter->filterAlertList($list);
+            }
+        }
+        $pagination = $list->getPagination();
+        $alerts = $pagination->getCurrentPageResults();
+        if (!$alerts) {
+            $alerts = [];
+        }
+
         $this->set('items', $alerts);
+        $this->set('filterValues', $filterValues);
         $this->set('token', $this->app->make('token'));
+        $this->set('pagination', $pagination);
+    }
 
+    public function action_reload_results()
+    {
+        $b = $this->getBlockObject();
+        $bv = new BlockView($b);
+        $bv->render('view');
+        exit;
```

Comment:
This shouldn't exit, it should return a response

Reply:
That doesn't work, unfortunately – it returns a response twice. It's something deeper that is core related.